skimage/feature/_hessian_det_appx.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+
+
+def _clip(x, low, high):
+    """Clips coordinate between high and low.
+
+    This method was created so that `hessian_det_appx` does not have to make
+    a Python call.
+
+    Parameters
+    ----------
+    x : int
+        Coordinate to be clipped.
+    low : int
+        The lower bound.
+    high : int
+        The higher bound.
+
+    Returns
+    -------
+    x : int
+        `x` clipped between `high` and `low`.
+    """
+
+    if x > high:
+        return high
+    if x < low:
+        return low
+    return x
+
+
+def _integ(img, r, c, rl, cl):
+    """Integrate over the integral image in the given window
+
+    This method was created so that `hessian_det_appx` does not have to make
+    a Python call.
+
+    Parameters
+    ----------
+    img : array
+        The integral image over which to integrate.
+    r : int
+        The row number of the top left corner.
+    c : int
+        The column number of the top left corner.
+    rl : int
+        The number of rows over which to integrate.
+    cl : int
+        The number of columns over which to integrate.
+
+    Returns
+    -------
+    ans : int
+        The integral over the given window.
+    """
+
+    r = _clip(r, 0, img.shape[0] - 1)
+    c = _clip(c, 0, img.shape[1] - 1)
+
+    r2 = _clip(r + rl, 0, img.shape[0] - 1)
+    c2 = _clip(c + cl, 0, img.shape[1] - 1)
+
+    ans = img[r, c] + img[r2, c2] - img[r, c2] - img[r2, c]
+    return max(0, ans)
+
+
+# pythran export _hessian_matrix_det(float64[:,:], float)
```

Comment:
if we have a single code path, I think casting to a numpy float is OK.

Reply:
OK